File: OpenCV/14_sift/cpp/main.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
int main(){ auto g=cv::imread("input.jpg",cv::IMREAD_GRAYSCALE); if(g.empty()) return 1; auto s=cv::SIFT::create(); std::vector<cv::KeyPoint> k; cv::Mat d,o; s->detectAndCompute(g,cv::noArray(),k,d); cv::drawKeypoints(g,k,o,cv::Scalar::all(-1),cv::DrawMatchesFlags::DRAW_RICH_KEYPOINTS); std::cout<<k.size()<<" keypoints\n"; cv::imwrite("output_sift.png",o); }

File: OpenCV/14_sift/python/main.py
import cv2
g=cv2.imread("input.jpg",cv2.IMREAD_GRAYSCALE); assert g is not None
sift=cv2.SIFT_create(); kp,des=sift.detectAndCompute(g,None); out=cv2.drawKeypoints(g,kp,None,flags=cv2.DRAW_MATCHES_FLAGS_DRAW_RICH_KEYPOINTS); print(len(kp),des.shape if des is not None else None); cv2.imwrite("output_sift.png",out)
